src/Core/SqlManager/Form/SqlRequestFormDataHandler.php

```diff
@@ -0,0 +1,213 @@
+
+            $this->commandBus->handle($command);
+
+            $this->hookDispatcher->dispatchWithParameters('actionSqlRequestSave', [
+                'id' => $sqlRequestId,
+                'data' => $sqlRequestFormData,
+            ]);
+        } catch (SqlRequestException $e) {
+            $errors[] = $this->handleException($e);
+        }
+
+        return $errors;
+    }
+
+    /**
+     * @param array $requestSqlData
+     *
+     * @return AddSqlRequestCommand
+     *
+     * @throws SqlRequestConstraintException
+     */
+    private function getAddRequestSqlCommand(array $requestSqlData)
+    {
+        return new AddSqlRequestCommand(
+            $requestSqlData['name'],
+            $requestSqlData['sql']
+        );
+    }
+
+    /**
+     * @param int $sqlRequestId
+     * @param array $requestSqlData
+     *
+     * @return EditSqlRequestCommand
+     *
+     * @throws SqlRequestConstraintException
+     */
+    private function getEditRequestSqlCommand($sqlRequestId, array $requestSqlData)
+    {
+        return (new EditSqlRequestCommand(new SqlRequestId($sqlRequestId)))
+            ->setName($requestSqlData['name'])
+            ->setSql($requestSqlData['sql']);
+    }
+
+    /**
+     * Transform exception into translatable errors.
+     *
+     * @param SqlRequestException $e
+     *
+     * @return array Errors
+     */
+    private function handleException(SqlRequestException $e)
+    {
+        $exceptionType = get_class($e);
+
+        if (SqlRequestConstraintException::class === $exceptionType) {
+            return $this->getConstraintError($e);
+        }
+
+        return $this->getErrorByExceptionType($e);
+    }
+
+    /**
+     * Get error for constraint exception.
+     *
+     * @param SqlRequestConstraintException $e
+     *
+     * @return array
+     */
+    private function getConstraintError(SqlRequestConstraintException $e)
+    {
+        $invalidFieldDictionary = [
+            SqlRequestConstraintException::INVALID_NAME => 'name',
+            SqlRequestConstraintException::INVALID_SQL_QUERY => 'sql',
+            SqlRequestConstraintException::MALFORMED_SQL_QUERY => 'sql',
+        ];
+
+        $code = $e->getCode();
+
+        if (isset($invalidFieldDictionary[$code])) {
+            return [
+                'key' => 'The %s field is invalid.',
+                'parameters' => [
+                    $invalidFieldDictionary[$code],
+                ],
+                'domain' => 'Admin.Notifications.Error',
+            ];
+        }
+
+        return [
+            'key' => 'Invalid data supplied.',
+            'parameters' => [],
+            'domain' => 'Admin.Notifications.Error',
+        ];
+    }
+
+    /**
+     * Get error for exception.
+     *
+     * @param SqlRequestException $e
+     *
+     * @return array
+     */
+    private function getErrorByExceptionType(SqlRequestException $e)
+    {
+        $exceptionDictionary = [
```

Comment:
i dont really know if @eternoendless would accept that. :/

Reply:
I actually don't think returning an `$errors` array is goo practice
If I get it right this means if the creation went well you return the ID, but if there are errors you return an array?
but if it's an update and everything went fine you return neither? or an empty array?
it would be cleaner to simply throw exceptions and manage them elsewhere, this clarifies the behavior of the interface
Besides managing the Exception transformation in this class gives it too many responsibilities